crates/interpreter/src/instructions/macros.rs

```diff
@@ -189,8 +189,14 @@ macro_rules! as_u64_saturated {
 }
 
 macro_rules! as_usize_saturated {
-    ($v:expr) => {
-        as_u64_saturated!($v) as usize
+    ($interp:expr, $v:expr) => {
+        match ::core::convert::TryInto::<usize>::try_into(as_u64_saturated!($v)) {
+            Ok(val) => val,
+            Err(_) => {
+                $interp.instruction_result = InstructionResult::InvalidOperandOOG;
+                return;
+            }
+        }
```

Comment:
We should return saturated usize here, it is a consensus error if we fail here.

```suggestion
        ::core::convert::TryInto::<usize>::try_into(as_u64_saturated!($v)).unwrap_or(usize::MAX)
```

Reply:
@Wollac this can be a reason for test failure.

Reply:
> @Wollac this can be a reason for test failure.

I think so, I misunderstood of this macro.